pkg/networkpath/traceroute/runner.go

```diff
@@ -0,0 +1,217 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package traceroute
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sort"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/Datadog/dublin-traceroute/go/dublintraceroute/probes/probev4"
+	"github.com/Datadog/dublin-traceroute/go/dublintraceroute/results"
+
+	"github.com/google/uuid"
+)
+
+// TODO: are these good defaults?
+const (
+	DefaultSourcePort   = 12345
+	DefaultDestPort     = 33434
+	DefaultNumPaths     = 1
+	DefaultMinTTL       = 1
+	DefaultMaxTTL       = 30
+	DefaultDelay        = 50 //msec
+	DefaultReadTimeout  = 3 * time.Second
+	DefaultOutputFormat = "json"
+)
+
+// RunTraceroute wraps the implementation of traceroute
+// so it can be called from the different OS implementations
+//
+// This code is experimental and will be replaced with a more
+// complete implementation.
+func RunTraceroute(cfg Config) (NetworkPath, error) {
+	rawDest := cfg.DestHostname
+	dests, err := net.LookupIP(rawDest)
+	if err != nil || len(dests) == 0 {
+		return NetworkPath{}, fmt.Errorf("cannot resolve %s: %v", rawDest, err)
+	}
+
+	//TODO: should we get smarter about IP address resolution?
+	// if it's a hostname, perhaps we could run multiple traces
+	// for each of the different IPs it resolves to up to a threshold?
+	// use first resolved IP for now
+	dest := dests[0]
+
+	dt := &probev4.UDPv4{
+		Target:     dest,
+		SrcPort:    uint16(DefaultSourcePort), // TODO: what's a good value?
+		DstPort:    uint16(DefaultDestPort),   // TODO: what's a good value?
+		UseSrcPort: false,
+		NumPaths:   uint16(DefaultNumPaths),
+		MinTTL:     uint8(DefaultMinTTL), // TODO: what's a good value?
+		MaxTTL:     uint8(15),
```

Comment:
```suggestion
		MaxTTL:     uint8(DefaultMaxTTL),
```

MaxTTL of 15 is a bit low, can we use DefaultMaxTTL of 30 instead?

With TTL 15, it's sometimes not enough to reach final hop.